.github/workflows/test.yml

```diff
@@ -7,48 +7,31 @@ on:
       - master
 jobs:
   tests:
-    name: Unit tests
+    name: Sub modules test and code coverage
     runs-on: ubuntu-latest
-    timeout-minutes: 15
     steps:
       - name: Install Go
         uses: actions/setup-go@v2.1.3
         with:
           go-version: 1.18
-      - name: Checkout code
-        uses: actions/checkout@v2
-      - name: run tests
-        run: make test
-        if: env.GIT_DIFF
-      - name: run experimental tests
-        run: EXPERIMENTAL=true make test
-        if: env.GIT_DIFF
-  code_cov:
-    name: Cover report
-    needs: tests
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        envs: ["stable-codecov", "experimental-codecov"]
-    steps:
-      - name: Install Go
-        uses: actions/setup-go@v2.1.3
+        - uses: technote-space/get-diff-action@v6.0.1
+        id: git_diff
         with:
-          go-version: 1.18
+          PATTERNS: |
+            **/**.go
+            go.mod
+            go.sum
```

Comment:
I like the idea here but we should make sure tests are still required and this prevents us from doing so. Maybe this could be used for some of the other jobs such as sims (the one's that are not nightly) ?

Reply:
> I like the idea here but we should make sure tests are still required and this prevents us from doing so. Maybe this could be used for some of the other jobs such as sims (the one's that are not nightly) ?

This change does not block required checks. If there are no changes in go code, this action skips `sub-modules test and codecov`, and moves to the next step.